skl2onnx/algebra/onnx_operator_mixin.py

```diff
@@ -0,0 +1,168 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from sklearn.base import BaseEstimator
+from onnx import shape_inference
+from ..common._registration import get_converter, get_shape_calculator
+from ..common._topology import Variable
+from .._supported_operators import sklearn_operator_name_map
+from .onnx_operator import OnnxOperator
+from .type_helper import guess_initial_types
+
+
+class OnnxOperatorMixin:
+    """
+    Base class for *scikit-learn* operators
+    sharing an API to convert object to *ONNX*.
+    """
+
+    def to_onnx(self, X=None, name=None):
+        """
+        Converts the model in *ONNX* format.
+        It calls method *_to_onnx* which must be
+        overloaded.
+
+        :param X: training data, at least one sample,
+            it is used to guess the type of the input data.
+        :param name: name of the model, if None,
+            it is replaced by the the class name.
+        """
+        from .. import convert_sklearn
+        if name is None:
+            name = self.__class__.__name__
+        if X is None:
+            initial_types = self.infer_initial_types()
+        else:
+            initial_types = guess_initial_types(X, None)
+        return convert_sklearn(self, initial_types=initial_types)
+
+    def infer_initial_types(self):
+        """
+        Infers initial types.
+        """
+        if hasattr(self, 'enumerate_initial_types'):
+            return list(self.enumerate_initial_types())
+        raise RuntimeError("Method enumerate_initial_types is missing "
+                           "and initial_types are not defined")
+
+    def _find_sklearn_parent(self):
+        if hasattr(self.__class__, 'predict') and \
```

Comment:
() instead of `\`, replace at other places as well.

Reply:
Fixed.

Reply:
I see `\` on line 51 and 56.

Reply:
Fixed.